diff --color -Nur alex-2.3.5.orig/src/Main.hs alex-2.3.5.new/src/Main.hs
--- alex-2.3.5.orig/src/Main.hs	2019-05-12 00:09:28.010699824 +0800
+++ alex-2.3.5.new/src/Main.hs	2019-05-12 00:14:38.062694271 +0800
@@ -27,6 +27,9 @@
 #if __GLASGOW_HASKELL__ >= 610
 import Control.Exception ( bracketOnError )
 #endif
+#if __HUGS__
+import Control.Exception ( bracketOnError )
+#endif
 import Control.Monad ( when, liftM )
 import Data.Char ( chr )
 import Data.List ( isSuffixOf )
@@ -201,36 +204,17 @@
 -- WORDS_BIGENDIAN (see GenericTemplate.hs).
 
 always_imports :: String
-always_imports = "#if __GLASGOW_HASKELL__ >= 603\n" ++
-		 "#include \"ghcconfig.h\"\n" ++
-		 "#elif defined(__GLASGOW_HASKELL__)\n" ++
-		 "#include \"config.h\"\n" ++
-		 "#endif\n" ++
-		 "#if __GLASGOW_HASKELL__ >= 503\n" ++
-		 "import Data.Array\n" ++
+always_imports = "import Data.Array\n" ++
 		 "import Data.Char (ord)\n" ++
-		 "import Data.Array.Base (unsafeAt)\n" ++
-		 "#else\n" ++
-		 "import Array\n" ++
-		 "import Char (ord)\n" ++
-		 "#endif\n"
+		 "import Data.Array.Base (unsafeAt)\n"
 
 import_glaexts :: String
-import_glaexts = "#if __GLASGOW_HASKELL__ >= 503\n" ++
-		 "import GHC.Exts\n" ++
-		 "#else\n" ++
-		 "import GlaExts\n" ++
-		 "#endif\n"
+import_glaexts = ""
 
 import_debug :: String
-import_debug   = "#if __GLASGOW_HASKELL__ >= 503\n" ++
-		 "import System.IO\n" ++
+import_debug   = "import System.IO\n" ++
 		 "import System.IO.Unsafe\n" ++
-		 "import Debug.Trace\n" ++
-		 "#else\n" ++
-		 "import IO\n" ++
-		 "import IOExts\n" ++
-		 "#endif\n"
+		 "import Debug.Trace\n"
 
 templateDir :: IO FilePath -> [CLIFlags] -> IO FilePath
 templateDir def cli
@@ -334,6 +318,7 @@
 dieAlex :: String -> IO a
 dieAlex s = getProgramName >>= \prog -> die (prog ++ ": " ++ s)
 
+#ifndef __HUGS__
 #if __GLASGOW_HASKELL__ < 610
 bracketOnError
 	:: IO a		-- ^ computation to run first (\"acquire resource\")
@@ -349,3 +334,4 @@
     return r
  )
 #endif
+#endif
