--- alex-2.3.5/Setup.lhs	2010-11-26 18:44:57.000000000 +0800
+++ alex-2.3.5n/Setup.lhs	2019-05-11 23:38:54.473732659 +0800
@@ -30,7 +30,7 @@
 myPostBuild _ flags _ lbi = do
   let runProgram p = rawSystemProgramConf (buildVerbose flags) p (withPrograms lbi)
       cpp_template src dst opts = do
-        runProgram ghcProgram (["-o", dst, "-E", "-cpp", "templates" </> src] ++ opts)
+        runProgram cpphsProgram (["-o", dst, "-E", "--cpp", "templates" </> src] ++ opts)
 	runProgram perlProgram ["-i.bak", "-pe", crazy_perl_regexp, dst]
 
   sequence_ ([ cpp_template "GenericTemplate.hs" dst opts | (dst,opts) <- templates ] ++
