diff --color -Nur process/System/Process/Internals.hs processn/System/Process/Internals.hs
--- process/System/Process/Internals.hs	2007-11-03 22:43:54.000000000 +0800
+++ processn/System/Process/Internals.hs	2019-05-10 00:43:14.962779014 +0800
@@ -54,7 +54,7 @@
 import GHC.IOBase	( haFD, FD, Exception(..), IOException(..) )
 import GHC.Handle 	( stdin, stdout, stderr, withHandle_ )
 # elif __HUGS__
-import Hugs.Exception	( Exception(..), IOException(..) )
+import Hugs.Exception	( IOException(..) )
 # endif
 
 import Control.Concurrent
@@ -405,8 +405,7 @@
 withFilePathException :: FilePath -> IO a -> IO a
 withFilePathException fpath act = handle mapEx act
   where
-    mapEx (IOException (IOError h iot fun str _)) = ioError (IOError h iot fun str (Just fpath))
-    mapEx e                                       = throwIO e
+    mapEx (IOError h iot fun str _) = ioError (IOError h iot fun str (Just fpath))
 
 #if !defined(mingw32_HOST_OS) && !defined(__MINGW32__)
 withCEnvironment :: [(String,String)] -> (Ptr CString  -> IO a) -> IO a
diff --color -Nur process/System/Process.hs processn/System/Process.hs
--- process/System/Process.hs	2007-11-03 22:43:54.000000000 +0800
+++ processn/System/Process.hs	2019-05-10 00:47:45.078771693 +0800
@@ -34,6 +34,7 @@
 -}
 
 module System.Process (
+# ifdef __GLASGOW_HASKELL__
 	-- * Running sub-processes
 	ProcessHandle,
 	runCommand,
@@ -45,6 +46,7 @@
 	waitForProcess,
 	getProcessExitCode,
 	terminateProcess,
+#endif
  ) where
 
 import Prelude
@@ -57,9 +59,9 @@
 import System.Exit	( ExitCode(..) )
 
 import System.Posix.Internals
+# ifdef __GLASGOW_HASKELL__
 import GHC.IOBase	( FD )
 import GHC.Handle 	( fdToHandle' )
-
 -- ----------------------------------------------------------------------------
 -- runCommand
 
@@ -326,3 +328,4 @@
   c_waitForProcess
 	:: PHANDLE
 	-> IO CInt
+#endif
